### PR TITLE
added text mask helper for inputs

### PR DESCRIPTION
**Pull request by Nathan-O**
This adds the Text-Masking helper to the library.

### CODE REVIEW

**Comment by chadzink** [comment]
In the process of testing this

File: addon/components/bootstrap-mask-input.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      this.sendAction('key-press', ...arguments);
+    },
+
+    keyUp: function() {
+      this.sendAction('key-up', ...arguments);
+    },
+
+    keyDown: function() {
+      this.sendAction('key-down', ...arguments);
+    },
```
Comment (Samsinite) [inline addon/components/bootstrap-mask-input.js]: I don't see one of these for focusOut, but it if it working atm as it is, then these might not be needed anymore.

File: addon/components/bootstrap-mask-input.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  placeholder: '',
+  keepCharPositions: true,
+
+  guid: Ember.computed.readOnly('_guid'),
+  _guid: null,
+
+  keyPress: computedActionKey('key-press'),
+  keyUp: computedActionKey('key-up'),
+  keyDown: computedActionKey('key-down'),
+  focusOut: null,
```
Comment (Samsinite) [inline addon/components/bootstrap-mask-input.js]: Ditto, should this be a `computedActionKey`?

**Comment by chadzink** [comment]
The Usage section of the Readme.md is a bit off format.
https://github.com/wildland/ember-bootstrap-controls/blob/text-masking-addon/README.md

**Comment by Nathan-O** [comment]
@chadzink the Freestyle component is implemented,  & @Samsinite the working freestyle does Not have focusOut applied to it and no errors (Note: I tried it with focusOut registered as a Fn, and that threw a few errors, just FYI)

File: addon/components/bootstrap-mask-input.js (hunk 1 above)
Comment (chadzink) [inline addon/components/bootstrap-mask-input.js]: Without these I get an error when testing

**Comment by Samsinite** [comment]
Updated this to use *dasherized* attrs so that they don't conflict with default component events. Want to take another look guys?

**Comment by Nathan-O** [comment]
👍  Sam's changes look good to me (Can't click approve review on my own PRs, so hope this'll count as my green light)

**Comment by chadzink** [comment]
:+1: Tested with dasherized attrs, looks good

**Pull request merged**